crates/nu-protocol/src/engine/engine_state.rs

```diff
@@ -1007,6 +1007,10 @@ impl EngineState {
             .map(|d| d.as_string().unwrap_or_default())
             .unwrap_or_default()
     }
+
+    pub fn get_file_contents(&self) -> Vec<(Vec<u8>, usize, usize)> {
+        self.file_contents.clone()
```

Comment:
Can we get away here without cloning?

Reply:
Agreed, you'll be copying the entire source ever read by Nushell just to find sudo, that seems like an overkill. Most likely you can return `&[u8]`.